docs/specs/clients/sign/data-structures.md

```diff
@@ -196,3 +200,162 @@ Namespace Config is set within the setNamespaceConfig method that is a part of t
   }
 }
 ```
+
+## Wallet Authentication Data Structures
+
+### Miscellaneous
+
+```typescript
+interface Participant {
+  publicKey: string;
+  metadata: Metadata;
+}
+
+interface SessionAuthenticateRequest {
+  id: number;
+  topic: string;
+  params: {
+    requester: Participant;
+    authPayload: PayloadParams;
+    expiryTimestamp: number;
+  };
+}
+```
+
+### Request Params
+
+```typescript
+interface RequestParams {
+  chains: string[];
+  domain: string;
+  nonce: string;
+  aud?: string;
+  type?: string;
+  nbf?: string;
+  exp?: string;
+  statement?: string;
+  requestId?: string;
+  resources?: string[];
+  pairingTopic?: string;
+  methods?: string[];
+}
+```
+
+### Respond Params
+
+```typescript
+type RespondParams = ResultResponse | ErrorResponse;
+```
+
+### Payload Params (partial Cacao)
+
+Used for requester to authenticate wallet
+
+```typescript
+interface PayloadParams {
+  type: string; // same as Cacao Header type (t)
+  chains: string[];
+  domain: string;
+  aud: string;
+  version: string;
+  nonce: string;
+  iat: string;
```

Comment:
If this is a part of the session authenticate request, we should remove `iat`. Wallet should set it.

Reply:
@ganchoradkov @llbartekll